robottelo/cli/factory.py

```diff
@@ -1462,9 +1462,14 @@ def make_lifecycle_environment(options=None):
         -h, --help                  print help
 
     """
-    # Organization ID is required
-    if not options or not options.get('organization-id', None):
-        raise CLIFactoryError('Please provide a valid ORG ID.')
+    # Organization Name, Label or ID is a required field.
+    if (
+            not options or
+            not options.get('organization', None) and
+            not options.get('organization-label', None) and
+            not options.get('organization-id', None)):
```

Comment:
It is possible for a user to supply a value that evaluates to false. For example, `bool(0)` evaluates to `False`. I think it is simpler and more robust to check for the existence of keys than examining values.

```python
if (
    not options or
    'organization' not in options and
    'organization-label' not in options and
    'organization-id' not in options)):
```